File: Java/680.java
class Solution {
    public boolean validPalindrome(String s) {
        int n = s.length();
        int i = 0, j = n-1, count = 0;
        while(i<j){
            if(s.charAt(i)!=s.charAt(j)){
                if(s.charAt(i)==s.charAt(j-1)){
                    j--;
                    count++;
                }
                else if(s.charAt(i+1)==s.charAt(j)){
                    i++;
                    count++;
                }
                else{
                    count += 2;
                    break;
                }
            }
            i++;
            j--;
        }
        if(count<=1) return true;
        i = 0; 
        j = n-1; 
        count = 0;
        while(i<j){
            if(s.charAt(i)!=s.charAt(j)){
                if(s.charAt(i+1)==s.charAt(j)){
                    i++;
                    count++;
                }
                else if(s.charAt(i)==s.charAt(j-1)){
                    j--;
                    count++;
                }
                else{
                    count += 2;
                    break;
                }
            }
            i++;
            j--;
        }
        if(count<=1) return true;
        return false;
    }
}

File: Python/680.py
class Solution(object):
    def validPalindrome(self, s):
        n = len(s)
        i = 0
        j = n-1
        count = 0
        while i<j:
            if s[i] != s[j]:
                if s[i+1] == s[j]:
                    count += 1
                    i += 1
                elif s[i] == s[j-1]:
                    count += 1
                    j -= 1
                else:
                    count = 2
                    break
            i += 1
            j -= 1
        if count <= 1:
            return True
        i = 0
        j = n-1
        count = 0
        while i<j:
            if s[i] != s[j]:
                if s[i] == s[j-1]:
                    count += 1
                    j -= 1
                elif s[i+1] == s[j]:
                    count += 1
                    i += 1
                else:
                    count = 2
                    break
            i += 1
            j -= 1
        if count <= 1:
            return True
        return False